src/WebJobs.Script/Description/DotNet/CSharp/CSharpCompilation.cs

```diff
@@ -62,12 +64,23 @@ private static bool IsOrUsesAssemblyType(ITypeSymbol typeSymbol, IAssemblySymbol
 
         public void Emit(Stream assemblyStream, Stream pdbStream, CancellationToken cancellationToken)
         {
-            EmitResult result = _compilation.Emit(assemblyStream, pdbStream, cancellationToken: cancellationToken);
+            var compilationWithAnalyzers = _compilation.WithAnalyzers(GetAnalyzers());
+            var diagnostics = compilationWithAnalyzers.GetAnalyzerDiagnosticsAsync().Result;
+            var emitResult = compilationWithAnalyzers.Compilation.Emit(assemblyStream, pdbStream, cancellationToken: cancellationToken);
 
-            if (!result.Success)
+            diagnostics = diagnostics.AddRange(emitResult.Diagnostics);
+
+            if (diagnostics.Any(di => di.Severity == DiagnosticSeverity.Error))
             {
-                throw new CompilationErrorException("Script compilation failed.", result.Diagnostics);
-            }
+                throw new CompilationErrorException("Script compilation failed.", diagnostics);
+            }            
+        }
+
+        private static ImmutableArray<DiagnosticAnalyzer> GetAnalyzers()
+        {
+            // Simply getting the built in analyzers for now.
+            // This should eventually be enhanced to dynamically discover/load analyzers.
```

Comment:
Not sure at this point, but it's conceivable that extensions may want to add their analyzers down the road. Just a thought.